tdh/resource_cluster_backup.go

```diff
@@ -0,0 +1,302 @@
+			"description": schema.StringAttribute{
+				Description: "Description for the create backup",
+				Optional:    true,
+			},
+			"service_type": schema.StringAttribute{
+				Description: "Service Type for Backup",
+				Optional:    true,
+			},
+		},
+	}
+
+}
+
+// Create a new resource
+func (r *clusterBackupResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	tflog.Info(ctx, "INIT__CreateBackup")
+	// Retrieve values from plan
+	var plan ClusterBackupResourceModel
+	diags := req.Plan.Get(ctx, &plan)
+	if resp.Diagnostics.Append(diags...); resp.Diagnostics.HasError() {
+		return
+	}
+
+	resp.Diagnostics.Append(req.Config.Get(ctx, &plan)...)
+	query := &controller.ClusterBackupQuery{
+		ID: plan.ClusterID.ValueString(),
+	}
+	fmt.Println(query)
+	// Generate API request body from plan
+	request := controller.BackupCreateRequest{
+		ClusterId:      plan.ClusterID.ValueString(),
+		Name:           plan.Name.ValueString(),
+		Description:    plan.Description.ValueString(),
+		BackupSchedule: "ON_DEMAND",
+		BackupType:     "FULL",
+		ServiceType:    plan.ServiceType.ValueString(),
+	}
+
+	tflog.Info(ctx, "req param", map[string]interface{}{"create-backup-request": request})
+
+	response, err := r.client.Controller.CreateClusterBackup(&request)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Submitting request",
+			"Could not create Cluster Backup , unexpected error: "+err.Error(),
+		)
+		return
+	}
+	//Check if the response is nil or empty
+	if response == nil {
+		resp.Diagnostics.AddError(
+			"Unexpected response",
+			"The API response for creating a cluster backup is empty",
+		)
+	}
+
+	tflog.Info(ctx, "INIT__Fetching Cluster Backup")
+	//var backups model.Paged[model.ClusterBackup]
+
+	backups, err := r.client.Controller.GetBackups(&controller.BackupQuery{
+		ServiceType: request.ServiceType,
+		Name:        request.Name,
+	})
+	if err != nil {
+		resp.Diagnostics.AddError("Fetching cluster backups",
+			"Could not fetch cluster bakups by name, unexpected error: "+err.Error(),
+		)
+		return
+	}
+
+	if backups.Page.TotalElements == 0 {
+		resp.Diagnostics.AddError("Fetching Cluster Backup",
+			fmt.Sprintf("Could not find any cluster backup by backup name [%s], server error must have occurred while creating user.", plan.Name.ValueString()),
+		)
+		return
+	}
+
+	// Map response body to schema and populate Computed attribute values
+	createdBackup := &(*backups.Get())[0]
+
+	// Map response body to schema and populate Computed attribute values
+	if saveBackupFromResponse(&ctx, &plan, createdBackup) != 0 {
+		return
+	}
+
+	// Set state to fully populated data
+	diags = resp.State.Set(ctx, plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	tflog.Info(ctx, "END__CreateBackup")
+}
+
+func (r *clusterBackupResource) Update(ctx context.Context, request resource.UpdateRequest, resp *resource.UpdateResponse) {
+	tflog.Info(ctx, "INIT__Delete")
```

Comment:
change message, and return early from method in error saying update not supported